### PR TITLE
Jetpack Connect: Auto authorize on signup via query

### PR DESCRIPTION
**Pull request by sirreal**
When creating an account, an action is dispatched:

https://github.com/Automattic/wp-calypso/blob/62ff54cb613b114ec8fe46e7a999a96da83654fc/client/jetpack-connect/auth-logged-out-form.jsx#L54

https://github.com/Automattic/wp-calypso/blob/62ff54cb613b114ec8fe46e7a999a96da83654fc/client/state/jetpack-connect/actions.js#L323-L343

Which has the side effect of setting `autoAuthorize` state:

https://github.com/Automattic/wp-calypso/blob/62ff54cb613b114ec8fe46e7a999a96da83654fc/client/state/jetpack-connect/reducer/jetpack-connect-authorize.js#L82-L103

Which, in turn, will autoAuthorize:

https://github.com/Automattic/wp-calypso/blob/62ff54cb613b114ec8fe46e7a999a96da83654fc/client/jetpack-connect/auth-logged-in-form.jsx#L101-L108

---

Rather than doing that, let's just attach a query param saying we've authorized!

`auth_approved=true`.

This PR removes a query param `new_user_started_connection ` that was apparently unused and set in an old user creation flow (@johnHackworth confirm 🙏 )

## Testing
1. Start logged out from WordPress.com
1. Make a new Jetpack site
1. Connect the site
1. Create a new user from the connection screen
1. You should be moved to the authorize step with an `auth_approved` query parameter.
1. Authorize should proceed without requiring user interaction.

### CODE REVIEW

File: client/jetpack-connect/auth-logged-out-form.jsx [hunk 1]
```diff
@@ -108,7 +104,10 @@ class LoggedOutForm extends Component {
 				{ this.renderLocaleSuggestions() }
 				<AuthFormHeader authQuery={ this.props.authQuery } />
 				<SignupForm
-					redirectToAfterLoginUrl={ this.getRedirectAfterLoginUrl() }
+					redirectToAfterLoginUrl={ addQueryArgs(
+						{ already_authorized: true },
```
Comment (sirreal) [inline client/jetpack-connect/auth-logged-out-form.jsx]: This is not the query param you want 😕

**Comment by sirreal** [comment]
Tests need cleanup.

**Comment by sirreal** [comment]
Rebased and ready for testing/review!

File: client/jetpack-connect/auth-logged-out-form.jsx [hunk 2]
```diff
@@ -108,7 +112,7 @@ class LoggedOutForm extends Component {
 				{ this.renderLocaleSuggestions() }
 				<AuthFormHeader authQuery={ this.props.authQuery } />
 				<SignupForm
-					redirectToAfterLoginUrl={ this.getRedirectAfterLoginUrl() }
+					redirectToAfterLoginUrl="!! THIS PROP IS REQUIRED BUT UNUSED !!"
```
Comment (sirreal) [inline client/jetpack-connect/auth-logged-out-form.jsx]: Yeah… worth having a look at what's going on here.

See p1513354730000161-slack-jetpack-conn-services

**Comment by johnHackworth** [comment]
> This PR removes a query param new_user_started_connection that was apparently unused and set in an old user creation flow (@johnHackworth confirm :pray: 

confirmed!! this looks good to me :+1:

File: client/jetpack-connect/auth-logged-out-form.jsx (hunk 2 above)
Comment (sirreal) [reply client/jetpack-connect/auth-logged-out-form.jsx]: Good idea 👍 

I've been looking into how other parts of Calypso do this but it's not clear to me yet. Hopefully we can make this a bit less confusing.

Comment (sirreal) [reply client/jetpack-connect/auth-logged-out-form.jsx]: Added a component-level docblock in 106537b, let me know if you think it needs work.

**Pull request merged**